### ISSUE TITLE
[BUG] Unable to select filters

### ISSUE THREAD

**Issue by Tim040** [V1P3R-FMG/free-map-genie#21]
Extension largely works as intended on iOS, using Orion Browser having installed the Firefox version if this extension, but I cannot click on the filters while the extension is active. Without the extension this does work.

Not too sure if this is an issue because I am using it from my phone

**Comment by MrFusiion** [comment]
Well filters should work for both the chrome and firefox versions. But because I don't own an iOS device I can't actively target/debug for this platform.

**Comment by BasedOnPixels** [comment]
I can't toggle any of the filters on firefox. I can mark objects on the map but I can't interact with the filters on the left in any way. Game I tried it on was Elden Ring.

![firefox_2023-11-15_18-00-11](https://github.com/V1P3R-FMG/free-map-genie/assets/13744383/0046861c-4ddd-44c8-a94e-fc264ed186f4)

**Comment by MrFusiion** [comment]
Well tested it on firefox-desktop-120.0b9, And it still seems to work for me.
Do the filters work occasionally or non at all?

### It would also be helpful to provide:
- errors/warnings
- platform (linux, windows, ...)
- browser version
- if applicable steps to reproduce the error.
- other information 'if you think it would be helpful, to much information is better than none'.

#### Errors for FMG are prefixed.
![image](https://github.com/V1P3R-FMG/free-map-genie/assets/71613276/a603ca63-b1fd-428a-8fba-800f28c39f2d)
#### Errors for mapgenie will have map.js as src.
![image](https://github.com/V1P3R-FMG/free-map-genie/assets/71613276/49c4931d-fd3f-4deb-a3d8-449401705de6)

**Comment by BasedOnPixels** [comment]
I'm running Firefox 119.0.1 (64bit) on Windows 11

When I open the site it refreshes like 4-5 times by itself or something, during which I'm able to turn filters on/off. Only after it stops I can't interact with it anymore.

I have turned off all my browser add-ons just in case but it didn't help either.

Errors/warnings I don't see any on the site itself, I'm not sure if the error from the console will help much.

https://puu.sh/JUzIo/02dec08d40.mp4

![image](https://github.com/V1P3R-FMG/free-map-genie/assets/13744383/e588129d-4ea4-4ff6-805d-9ff086006e8e)

![image](https://github.com/V1P3R-FMG/free-map-genie/assets/13744383/d544c26f-66c2-47c5-9351-e80ac7bc1fc0)

**Comment by MrFusiion** [comment]
The refreshing is because the extension can't communicate with the injected script.
One case I know this can happen is when permissions are not enabled for the extension.
And can't inject the content script to the page.

Just to make sure can you check if permissions are enabled for the extension.
Step 6 in firefox install guide.
[Firefox installation guide](https://github.com/V1P3R-FMG/free-map-genie/blob/main/docs/installation/firefox.md)

For now I maybe work on something that will try to detect this and warn the user instead of refreshing the page.

**Comment by MrFusiion** [comment]
One thing for sure is that I will remove the auto reload and instead change it to a warning/error that the extension couldn't find the `window.store` object.
And requests the user to reload the page manually.

**Comment by MrFusiion** [comment]
@BasedOnPixels It could be helpful for me if you could check this for me.

- Go to this Url in Firefox `about:debugging#/runtime/this-firefox`.
- And check if the background script is running under the fmg extension.
![image](https://github.com/V1P3R-FMG/free-map-genie/assets/71613276/033a940a-e51e-4bea-8bca-44797a7bb0f9)
If not go to a map page `elden ring` for example. And check if to switches from `stopped` -> `running`
- Also check if the extension has any errors
![image](https://github.com/V1P3R-FMG/free-map-genie/assets/71613276/dd8754a2-f3a3-4c01-b873-6c3f38630dea)
It shows a warning because of a invalid manifest value already working on fixing that.

**Comment by BasedOnPixels** [comment]
The background script is stopped on the about:debugging page. If I load it in as a temporary add-on, i can reload it and it is running at first but stops seemingly after a while.

The regular and the dev build didn't have any errors.

On the mapgenie.io page it tells me that window.store is already defined and asks me to reload the page.

**Comment by MrFusiion** [comment]
When loading the dev build you can click on the inspect button and it will open a dev tools window, go to the console tab.
And open a mapgenie map and if everything is right it should print a `block request`, `redirect request` and a `allow request`.

![image](https://github.com/V1P3R-FMG/free-map-genie/assets/71613276/a1d10fc1-25a6-4c8e-842b-1ac6b9c31738)

**Comment by MrFusiion** [comment]
> On the mapgenie.io page it tells me that window.store is already defined and asks me to reload the page.

By mapgenie.io page do you mean the homepage?
![image](https://github.com/V1P3R-FMG/free-map-genie/assets/71613276/0cc8ed1f-d628-46d5-809b-4842adb23114)
Because that should not be happening. If so open the dev tools on the homepage you can do this by pressing f12.
And type `window.store` in the console.
![image](https://github.com/V1P3R-FMG/free-map-genie/assets/71613276/26cad7ba-dc68-4fe8-9c25-d058078942ce)
It should return undefined.

**Comment by BasedOnPixels** [comment]
Oh no, i meant generally on mapgenie.io, as in the Elden Ring map, sorry for not being clear about it. On the main page it doesn't show the error and window.store returns undefined.

The console on the dev add-on doesn't say more than "Background script loaded".

![image](https://github.com/V1P3R-FMG/free-map-genie/assets/13744383/dabc6b56-6fb3-4b76-8edd-de696285b263)

**Comment by MrFusiion** [comment]
Ok so there is the problem the backgrounds script is unable to block the script hence why `window.store` is already defined.
Now to find out why its unable to block.

**Comment by BasedOnPixels** [comment]
I just figured it out. 

I had to enable "Access your data for all websites".
I thought turning on data for mapgenie.io was enough due to it only requiring access to the site?

![image](https://github.com/V1P3R-FMG/free-map-genie/assets/13744383/d4753a44-31fe-46e2-8196-133d9df315ef)


![image](https://github.com/V1P3R-FMG/free-map-genie/assets/13744383/4c216a80-19a9-47e8-8faf-63e2e8fe7255)

**Comment by MrFusiion** [comment]
Ok good, But does it fix the filters?

**Comment by MrFusiion** [comment]
![image](https://github.com/V1P3R-FMG/free-map-genie/assets/71613276/35b106b4-d767-471f-a0ed-fdc1e2600b57)
Maybe I can finally switch over to declarativeNetRequest for Firefox. Which I already use for the chrome version.

**Comment by BasedOnPixels** [comment]
It does fix the problem, I can set filters and they all save and load when I refresh the page. It seems to be completely working now after giving it full permission to all sites.

**Comment by MrFusiion** [comment]
Well just implemented `declarativeNetRequest` for the Firefox version, which doesn't need any extra permissions as far as I know.
As for the OP case its probably the same issue. So I will mark this as solved for now. It can always be reopened.

Solution:
> I just figured it out.
> 
> I had to enable "Access your data for all websites". I thought turning on data for mapgenie.io was enough due to it only requiring access to the site?
> ![image](https://user-images.githubusercontent.com/13744383/283385414-d4753a44-31fe-46e2-8196-133d9df315ef.png)

**Comment by Tim040** [comment]
I tried the new release and I get the new ‘window.store is already defined’ error pop-up, but I assume this is because the browser I am using on my phone does not allow to set the optional permissions. 

I think a fix for this could be to have these permissions set as required instead of optional, but that’s probably not a desirable solution.

**Comment by MrFusiion** [comment]
> I think a fix for this could be to have these permissions set as required instead of optional, but that’s probably not a desirable solution.

Its a manifest v3 issue which makes the content script injection optional on Firefox.

Well I found out what breaks the filter, The map.js script is loaded twice the default one because my background is unable to block it, and one after my extension is done with the setup face.
So one easy fix is to check if the store is already defined and not load the map a second time.
But some features may not work.
- Pro maps (tarkov)
- Mock user (not so important just login with your account)

A second solution is add a second Firefox version for manifest v2 which I don't think has the option permissions by default.

**Comment by MrFusiion** [comment]
Its possible that this issue is fixed by disabling the `declarative net request` option in the browser action.
If it still does not work reopen this thread or post a new issue.
